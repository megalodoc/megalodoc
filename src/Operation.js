import React from 'react'
import PropTypes from 'prop-types'

const Operation = ({verb, operation}) =>
  <div>
    <h3>{verb}</h3>
    <p>{operation.summary}</p>
    <p>{operation.description}</p>
    {/* render each responses object */}
  </div>

Operation.propTypes = {
  verb: PropTypes.string.isRequired,
  operation: PropTypes.shape({
    tags: PropTypes.array,
    summary: PropTypes.string,
    description: PropTypes.string,
    externalDocs: PropTypes.object,
    operationId: PropTypes.string,
    consumes: PropTypes.array,
    produces: PropTypes.array,
    parameters: PropTypes.array,
    responses: PropTypes.object,
    schems: PropTypes.array,
    deprecated: PropTypes.boolean,
    security: PropTypes.array
  }).isRequired
}

export default Operation
