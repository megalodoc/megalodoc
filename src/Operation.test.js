import React from 'react'
import renderer from 'react-test-renderer'
import Operation from './Operation'
import petstore from './fixtures/petstore.json'

it('renders the Operation component', () => {
  const paths = petstore.paths
  const path = Object.keys(paths)[0]
  const pathItem = paths[path]
  const verb = Object.keys(pathItem)[0]
  const operation = pathItem[verb]
  const component = renderer.create(<Operation verb={verb} operation={operation} />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
