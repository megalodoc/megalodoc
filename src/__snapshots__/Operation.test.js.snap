// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Operation component 1`] = `
<div>
  <h3>
    post
  </h3>
  <p>
    Add a new pet to the store
  </p>
  <p>
    
  </p>
</div>
`;
